extensions/incubator/src/main/java/io/opentelemetry/extension/incubator/trace/ExtendedSpan.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.incubator.trace;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanContext;
+
+/** Extended {@link Span} with experimental APIs. */
+public interface ExtendedSpan extends Span {
+
+  /**
+   * Adds a link to this {@code Span}.
+   *
+   * <p>Links are used to link {@link Span}s in different traces. Used (for example) in batching
+   * operations, where a single batch handler processes multiple requests from different traces or
+   * the same trace.
+   *
+   * <p>Implementations may ignore calls with an {@linkplain SpanContext#isValid() invalid span
+   * context}.
+   *
+   * <p>Callers should prefer to add links before starting the span via {@link
+   * SpanBuilder#addLink(SpanContext)} if possible.
+   *
+   * @param spanContext the context of the linked {@code Span}.
+   * @return this.
+   */
+  default Span addLink(SpanContext spanContext) {
+    return addLink(spanContext, Attributes.empty());
+  }
```

Comment:
I wish there was something in the description here that made the directionality of the link clearer. I'm concerned that a causal user might think that this adds a link _from_  the `SpanContext` _to_ this current `Span` on which this is being invoked. Maybe changing:

```
Adds a link to this {@code Span}.
```
to something like
```
Creates a new span link that points from this span to another span context. The resulting link is added to this span.
```
could make this clearer?

Sorry if it seems particularly nitpicky. Technically the link itself does live in this Span, so the implementation does "add a link to this span", but conceptually the link points _from_ this Span _to_ the span with the given context....right?

Reply:
Hmm.. I think a good test for whether this is a valid concern is feedback on whether the SpanBuilder#addLink methods are confusing, since those share the same semantics and javadoc. Those have been stable for years and I haven't heard any feedback - although they are admittedly not the most commonly used API feature. To me, the method name and javadoc make it clear that the resulting link will be attached to this span. I think a user would have to be willfully ignorant to reach a different conclusion 😁

Reply:
Ok, I am totally willing to relent on this directionality concern then. It does have parity with the other methods, and I think you'd have to try pretty hard to be confused by it (as you suggested)....but I've seen wackier things happen! 😁